### ISSUE TITLE
 Question in Conjugate Gradient method

### ISSUE THREAD

**Issue by Aditya-Tejaswi** [hkaggarwal/modl#11]
Hello Sir,

First of all, thank you for maintaining and sharing this repository. The idea of using the CG method for generating MR images is truly impressive :smiley:  . 

My query was related to the Conjugate Gradient methods. In the file ```model.py```, under the ```myCG``` function, at line 101, 
```python3 
alpha = rTr / tf.to_float(tf.reduce_sum(tf.conj(p)*Ap))
```
The ```tf.to_float()``` command discards the imaginary part. I checked the values over multiple steps and found that the imaginary values were very close to 0, so they could be discarded. It makes sense to discard the imaginary values when something like ```x*tf.conj(x)``` happens, as it would be = modulus(x)², which happens in line 105 when ```rtrNew``` is calculated. I'm just curious to know how did you reach the conclusion that these values would be small enough to be discarded :green_circle: . Did you also check the values or was it based on some matrix properties of ```Ap```? I read this topic in _'Numerical recipes for C'_ and it mentions that **A** should be **__symmetric and positive definite__**. But I don't know what properties ```Ap``` has which makes it behave the way it does in the code. 

I would highly appreciate your feedback :smile:
